### PR TITLE
Update gradle to 4.10.2

### PR DESCRIPTION
**Pull request by sciencewhiz**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,11 +56,14 @@ repositories {
 
 
 dependencies {
+    errorprone 'com.google.errorprone:error_prone_core:2.0.9'
     compile 'org.apache.velocity:velocity:1.7'
     compile 'org.yaml:snakeyaml:1.10'
     compile 'com.github.jgraph:jgraphx:v3.3.1.1'
-    compile 'org.projectlombok:lombok:1.16.18'
+	compile 'com.sun.activation:javax.activation:1.2.0'
```
Comment (AustinShalit) [inline build.gradle]: Extra space

File: settings.gradle [hunk 2]
```diff
@@ -0,0 +1 @@
+enableFeaturePreview('STABLE_PUBLISHING')
```
Comment (AustinShalit) [inline settings.gradle]: EOL @ EOF

File: build.gradle [hunk 3]
```diff
@@ -54,11 +56,14 @@ repositories {
 
 
 dependencies {
+    errorprone 'com.google.errorprone:error_prone_core:2.3.1'
     compile 'org.apache.velocity:velocity:1.7'
     compile 'org.yaml:snakeyaml:1.10'
     compile 'com.github.jgraph:jgraphx:v3.3.1.1'
-    compile 'org.projectlombok:lombok:1.16.18'
+    compile 'com.sun.activation:javax.activation:1.2.0'
```
Comment (JLLeitschuh) [inline build.gradle]: ## ⚠️ 🔥 ⚠️ Someone should read this licence!!!! ⚠️ 🔥 ⚠️ 

Let's make sure we're not coupling ourself to a licence we don't want.
https://search.maven.org/artifact/com.sun.activation/javax.activation/1.2.0/jar

Comment (SamCarlberg) [reply build.gradle]: Looks like Oracle's standard GPL with classpath exception.

Why is this required, though?

Comment (JLLeitschuh) [reply build.gradle]: @SamCarlberg Are you allowed to re-package and publish releases with oracle's jar's in them with their GPL with classpath exception licence?

Comment (sciencewhiz) [reply build.gradle]: It's deprecated in JDK 9, and the module isn't loaded by default. This seemed like the easiest and most future proof of oracle's migration suggestions.

Comment (SamCarlberg) [reply build.gradle]: Ah, looks like it's used for DnD. Looks nontrivial to migrate to something else so this is fine with me

Comment (JLLeitschuh) [reply build.gradle]: DnD?

Comment (SamCarlberg) [reply build.gradle]: Drag-and-drop

Comment (JLLeitschuh) [reply build.gradle]: You need to re-distribute this application with the oracle licence if you add this dependency. Just FYI.

**Pull request merged**